src/object_store/src/object/mod.rs

```diff
@@ -910,11 +910,21 @@ pub async fn build_remote_object_store(
             set your endpoint to the environment variable RW_S3_ENDPOINT.");
             panic!("Passing s3-compatible is not supported, please modify the environment variable and pass in s3.");
         }
-        minio if minio.starts_with("minio://") => ObjectStoreImpl::S3(
-            S3ObjectStore::with_minio(minio, metrics.clone(), config)
-                .await
-                .monitored(metrics),
-        ),
+        minio if minio.starts_with("minio://") => {
+            if std::env::var("RW_USE_OPENDAL_FOR_MINIO").is_ok() {
```

Comment:
minio and s3 are technically the same. How about reusing `RW_USE_OPENDAL_FOR_S3` instead or introducing a new env?

Reply:
Agree.